pkg/noun/manage.c

```diff
@@ -1005,6 +1005,10 @@ u3m_bail(u3_noun how)
     }
   }
 
+  // Reset the spin stack pointer
+  stk_u->off_w = u3R->off_w;
+  stk_u->fow_w = u3R->fow_w;
```

Comment:
We can't reset here with the inner road spin-stack values, it needs to be done with the values from the outer road. That could be all the time in `u3m_fall()`, or only on errors (at each `_setjmp()` callsite, the various u3m_soft_* functions). That reset logic also needs to be duplicated in `_cm_signal_recover()`, where we handle errors after `u3m_signal()` (ie, `siglongjmp()`).

Reply:
Note: Chatted with Joe, how this is done is fine because of how the values are set when the road gets created. I will still need to add this to `_cm_signal_recover()`

Reply:
Added this before the if statement in `_cm_signal_recover`